discord/guild.py

```diff
@@ -895,6 +896,65 @@ def get_member(self, user_id: int, /) -> Member | None:
         """
         return self._members.get(user_id)
 
+    _FETCHABLE = TypeVar(
+        "_FETCHABLE",
+        VoiceChannel,
+        TextChannel,
+        ForumChannel,
+        StageChannel,
+        CategoryChannel,
+        Thread,
+        Member,
+    )
+
+    async def get_or_fetch(
+        self, object_type: type[_FETCHABLE], id: int, /
+    ) -> _FETCHABLE | None:
+        """Shortcut method to get data from guild object if it's cached or fetch from api if it's not.
+
+        Parameters
+        ----------
+        object_type: :class:`Type[_FETCHABLE]`
```

Comment:
This typehint should be expanded in the docs. I'm making a PR for autodoc typehints, so this should be merged following that.

Reply:
The note below about the type can also be moved up here or just removed after the typehint is corrected.

Reply:
Removed not at all, made typehinting direct, it displays everything right that way.